ZDNA-prediction.local.2.ipynb

```diff
@@ -0,0 +1,459 @@
+    "model_confidence_threshold = None\n",
+    "minimum_sequence_length = None\n",
+    "check_sequence_variations = None\n",
+    "use_cuda_if_available = None"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@load_model_output.capture(clear_output=True)\n",
+    "def load_model(btn):\n",
+    "    global model_name, model_confidence_threshold, minimum_sequence_length, check_sequence_variations, model_file_path, tokenizer, model, use_cuda_if_available, is_cuda_available\n",
+    "    \n",
+    "    model_name = model_name_widget.value\n",
+    "    model_confidence_threshold = model_confidence_threshold_widget.value\n",
+    "    minimum_sequence_length = minimum_sequence_length_widget.value\n",
+    "    check_sequence_variations = check_sequence_variations_widget.value\n",
+    "    use_cuda_if_available = use_cuda_if_available_widget.value\n",
+    "    \n",
+    "    print('\\n\\ncompleted loading model\\n\\nmodel: {}\\nmodel confidence threshold: {}\\nminimum sequence length: {}'.format(model_name, model_confidence_threshold, minimum_sequence_length))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "load_model_button = widgets.Button(\n",
+    "    description='Load model',\n",
+    "    icon='truck-loading',\n",
+    ")\n",
+    "load_model_button.on_click(load_model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "_1qjq3i-VCEz"
+   },
+   "outputs": [],
+   "source": [
+    "@do_predictions_output.capture(clear_output=True)\n",
+    "def do_predictions(btn):\n",
+    "    uploaded_items = retrieve_input_files_list()\n",
+    "    \n",
+    "    #logging.basicConfig(\n",
+    "    #    filename='example.log',\n",
+    "    #    encoding='utf-8',\n",
+    "    #    format='%(asctime)s - %(levelname)s - %(name)s - %(message)s',\n",
+    "    #    level=logging.INFO\n",
+    "    #)\n",
+    "    \n",
+    "    zdnabert_model_downloader = ZdnabertModelDownloader()\n",
+    "    zdnabert_model_downloader.download_models(model_download_path)\n",
+    "    zdnabert_model_downloader.download_metas(model_download_path)\n",
+    "    \n",
+    "    zdnabert_model = ZdnabertModel(\n",
+    "        os.path.join(model_download_path, model_name),\n",
+    "        model_name=model_name,\n",
+    "        model_confidence_threshold=model_confidence_threshold,\n",
+    "        minimum_sequence_length=minimum_sequence_length,\n",
+    "        use_cuda_if_available=use_cuda_if_available,\n",
+    "    )\n",
+    "    \n",
+    "    sequence_variation_normal = SequenceVariationNormal()\n",
+    "    sequence_variation_reverse_complement = SequenceVariationReverseComplement()\n",
+    "    sequence_variations = [\n",
+    "        sequence_variation_normal,\n",
+    "        sequence_variation_reverse_complement,\n",
```

Comment:
missing if-statement so that `SequenceVariationReverseComplement` only gets used when  `check_sequence_variations` is `true`